File: factorial/factorial.php
//using php web

<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>
<body>
    <form action="algoritmos.php" method="POST">
        <input type="number" name = "fatorial" placeholder="Digite um número e retornarei o fatorial dele">
        <button type="submit">enviar</button>
    </form>
    <?php 
    
    function fat(){
        $num = $_POST['fatorial'];
        for ($n = $num - 1; $n != 1; $n--){
            $num *= $n;
        }
        echo  $num;
    }
    
    fat();
    
    ?>
</body>
</html>

File: factorial/factorial.c
#include <stdio.h>

void factorial(){
    int num = 0;
    printf("input a number and we'll tell you its factorial: "); 
    scanf("%d", &num); 
    for (int n = num - 1; n != 1; n--)
    {
        num *= n;
    }

    printf("%d", num);
}

int main(){
    
    factorial();

    return 0;
}